docker-compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: "3.0"
+
+services:
+  server:
+    build:
+      context: .
+    restart: always
+    ports:
+      - "7171:7171"
+      - "7172:7172"
+    depends_on:
+      - db
+    networks:
+      - db-network
+  db:
+    image: mysql
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: <your_root_password>
+      MYSQL_USER: forgottenserver
+      MYSQL_PASSWORD: <your_db_password>
+      # Automatically creates a database
+      MYSQL_DATABASE: forgottenserver
+    volumes:
+      # Saves all the data in ./db directory
+      - ./db:/var/lib/mysql
```

Comment:
I would use a volume here, instead of relying on the host file system.

Reply:
It's unusually hard to fish the files out of a docker-managed volume and the point here is for the data to persist, so I think this is the easy to understand option here.

Reply:
Why would you want to extract raw files from a database storage? It's not like you can do anything more with them out of the container.

Reply:
You can back them up and run a server out of them, see mariabackup etc.

Reply:
You can do the very same with volumes, just prepend `docker exec` to the same `mysqldump -u <user> -p <password> <db>` you would run.